Remove Quirk needsIPadMiniUserAgent for a series of domains

karlcow opened the pull request:
#### 5a4a1dc1c9c758f5d960bddc9f280fd0b85b1f48
<pre>
Remove Quirk needsIPadMiniUserAgent for a series of domains
<a href="https://bugs.webkit.org/show_bug.cgi?id=284103">https://bugs.webkit.org/show_bug.cgi?id=284103</a>
<a href="https://rdar.apple.com/140974340">rdar://140974340</a>

Reviewed by Brent Fulgham and Anne van Kesteren.

This list of domains had User Agent overrides to give an iPad Mini
UA instead of the usual desktop User Agent. This was done initially
to give most of the time UI compatible with touch events when the
website were unable to manage a device with a desktop UA and touch
interface. This list of domains has been tested and the UA override
is not required anymore.

We are also fencing hsbc domains behind a compile flag to be able
to test internally if we can reproduce the breakages that was occuring
for HSBC in the first place.

* <a href="https://rdar.apple.com/59480381">rdar://59480381</a> fidelity.com
* <a href="https://rdar.apple.com/51355671">rdar://51355671</a> tving.com
* <a href="https://rdar.apple.com/51355671">rdar://51355671</a> live.iqiyi.com
* <a href="https://rdar.apple.com/51355671">rdar://51355671</a> video.sina.com.cn
* <a href="https://rdar.apple.com/51355671">rdar://51355671</a> video.tudou.com
* <a href="https://rdar.apple.com/51355671">rdar://51355671</a> v.china.com.cn
* <a href="https://rdar.apple.com/49124313">rdar://49124313</a> trello.com
* <a href="https://rdar.apple.com/51463649">rdar://51463649</a> ted.com
* <a href="https://rdar.apple.com/67823088">rdar://67823088</a> nhl.com
* <a href="https://rdar.apple.com/50839844">rdar://50839844</a> jsfiddle.net

* Source/WTF/wtf/PlatformUse.h:
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsIPadMiniUserAgent):

Canonical link: <a href="https://commits.webkit.org/287705@main">https://commits.webkit.org/287705@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fb7d47cab75619385474ff026ca290849b0c0efc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/80584 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/59591 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/34517 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/85108 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/31566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/82695 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/68652 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/7897 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/62950 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/31566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83653 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/53061 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/73362 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/43255 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/50355 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/27521 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/30025 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/73564 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/71501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/28045 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/86541 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/79643 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7810 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/5511 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/71245 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/7985 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/69199 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/70485 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/17564 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/14494 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/13439 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/102050 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12473 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7772 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/13291 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/24844 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/7611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/11130 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/9416 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WTF/wtf/PlatformUse.h:
```diff
@@ -431,3 +431,9 @@
 #define USE_MODERN_AVCONTENTKEYSESSION 1
 #endif
+
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED < 150400) \
```
Actually, since this quirk is only relevant on iPad, we should remove the MAC bit here -- we don't ever need to trigger this Quirk on macOS! That will save us doing these lookups for Mac users, too.

brentfulgham reviewed (approved):
This change looks good, but please delete the `PLATFORM(MAC)` part of your `USE_` macro before landing -- we never need this to be true on macOS.

brentfulgham reviewed (approved):
Thank you for removing the macOS build target here. :-)

karlcow committed:
Remove Quirk needsIPadMiniUserAgent for a series of domains
https://bugs.webkit.org/show_bug.cgi?id=284103
rdar://140974340

Reviewed by Brent Fulgham and Anne van Kesteren.

This list of domains had User Agent overrides to give an iPad Mini
UA instead of the usual desktop User Agent. This was done initially
to give most of the time UI compatible with touch events when the
website were unable to manage a device with a desktop UA and touch
interface. This list of domains has been tested and the UA override
is not required anymore.

We are also fencing hsbc domains behind a compile flag to be able
to test internally if we can reproduce the breakages that was occuring
for HSBC in the first place.

* rdar://59480381 fidelity.com
* rdar://51355671 tving.com
* rdar://51355671 live.iqiyi.com
* rdar://51355671 video.sina.com.cn
* rdar://51355671 video.tudou.com
* rdar://51355671 v.china.com.cn
* rdar://49124313 trello.com
* rdar://51463649 ted.com
* rdar://67823088 nhl.com
* rdar://50839844 jsfiddle.net

* Source/WTF/wtf/PlatformUse.h:
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsIPadMiniUserAgent):

Canonical link: https://commits.webkit.org/287705@main
